subcommand/server-acl-init/command.go

```diff
@@ -332,6 +338,65 @@ service_prefix "" {
 		}
 	}
 
+	// Support ConnectInject using Kuberentes as an auth method
+	if c.flagCreateInjectAuthMethod {
+		// Get the Kubernetes service IP address
+		k8sService, err := clientset.CoreV1().Services(c.flagNamespace).Get("kubernetes", metav1.GetOptions{})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error getting kubernetes service: %s", err))
+			return 1
+		}
+
+		// Pull the CACert out of the kubeconfig we use to set up the k8s client
+
+		// Get auth method service account JWT
+		saName := fmt.Sprintf("%s-consul-connect-injector-authmethod-svc-account", c.flagReleaseName)
+		amServiceAccount, err := clientset.CoreV1().ServiceAccounts(c.flagNamespace).Get(saName, metav1.GetOptions{})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error getting service account: %s", err))
+			return 1
+		}
+
+		// Assume the jwt is the first secret attached to the service account
+		saSecret, err := clientset.CoreV1().Secrets(c.flagNamespace).Get(amServiceAccount.Secrets[0].Name, metav1.GetOptions{})
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Error getting service account JWT secret: %s", err))
+			return 1
+		}
+
+		// Set up auth method
+		aam := api.ACLAuthMethod{
+			Name: fmt.Sprintf("%s-consul-k8s-auth-method", c.flagReleaseName),
```

Comment:
You may want to add in a `Description` here like you do on the binding rule to indicate that consul-k8s seeded this for you.

Reply:
Added.